prov/util/src/util_coll.c

```diff
@@ -1004,23 +1012,18 @@ int ofi_join_collective(struct fid_ep *ep, fi_addr_t coll_addr,
 	av_set = container_of(set, struct util_av_set, av_set_fid);
 
 	if (coll_addr == FI_ADDR_NOTAVAIL) {
-		assert(av_set->av->coll_mc != NULL);
-		coll_mc = av_set->av->coll_mc;
+		assert(av_set->av->av_set);
+		coll_mc = &av_set->av->av_set->coll_mc;
 	} else {
 		coll_mc = (struct util_coll_mc*) ((uintptr_t) coll_addr);
 	}
+	/* TODO: This won't work if there are multiple ep's */
+	coll_mc->ep = ep;
```

Comment:
this problem isn't introduced by your changes, so i think the TODO is sufficient for now.

Could the ep be associated the with operation rather than the mc to avoid this problem?

Reply:
Yes, the ep needs to be associated with the operation.  I didn't look at what it would take to make that happen.

Reply:
Well, based on 1 minute of looking into this, moving the ep to the operation looks trivial.